### ISSUE TITLE
Action never completes after successfully snapping

### ISSUE THREAD

**Issue by nicolasbock** [canonical/action-build#44]
Hi,

After successfully completing and snapping the action gets stuck (for a lack of a better word) and is eventually timed out. This looks like the snapcraft process is waiting on something it is not seeing.

The latest build that exhibits this behavior can be found here:

https://github.com/nicolasbock/rabbitmq-server-snap/actions/runs/3177885875/jobs/5178819513#step:3:15178

Nick

**Comment by sergiusens** [comment]
Does it build fine without the action?

**Comment by nicolasbock** [comment]
Yes, locally I can build the snap in a VM or using LXD without such issues.

**Comment by ilya-fedin** [comment]
Having the same (kind of, logs stop in the middle of the build) problem at https://github.com/telegramdesktop/tdesktop after GitHub runners have updated [20230206.1](https://github.com/telegramdesktop/tdesktop/actions/runs/4201579338/jobs/7288721626) -> [20230217.1](https://github.com/telegramdesktop/tdesktop/actions/runs/4211688306/jobs/7310196259). The repo has action calling snapcraft manually, but [here](https://github.com/ilya-fedin/tdesktop/commit/6389ad4b173802f0514cfb44a8a1ede7d07c58a6) I tried to switch to action-build, sadly it [didn't](https://github.com/ilya-fedin/tdesktop/actions/runs/4322015859/jobs/7543920176) help.

**Comment by sergiusens** [comment]
Hi @ilya-fedin, I took a quick look and this looks more like running out of memory or the build taking too long.

The original issue was because of docker being provisioned on 22.04 which caused snapcraft to stall and do nothing as it did not detect any network. This was fixed in the action with some iptables rules.

**Comment by ilya-fedin** [comment]
@sergiusens if it runs out of memory, shouldn't the action stop? As for build taking too long, it was taking around 3 hours with 20230206.1 runner and now since 20230217.1 it times out after 6 hours, I don't really believe something can slow down it that much, it more looks like the connection with lxd container hangs after some time.

**Comment by sergiusens** [comment]
@ilya-fedin I found that explained on the runner-images repo https://github.com/actions/runner-images/issues/1918

There's a way to get a login shell into the runner to figure out what's happening, I don't have an instruction set handy, but I am certain @mr-cal does

**Comment by sergiusens** [comment]
I also have to walk back my comments on what this issue was about after reading the title and I cannot see the original linked log anymore :-(

**Comment by nicolasbock** [comment]
> The original issue was because of docker being provisioned on 22.04 which caused snapcraft to stall and do nothing as it did not detect any network. This was fixed in the action with some iptables rules.

Could you point me to where this is fixed @sergiusens ? My workflows are running on 20.04, not 22.04.

Since the old logs are gone, I restarted that exact workflow here:

https://github.com/nicolasbock/rabbitmq-server-snap/actions/runs/4348662023

**Comment by ilya-fedin** [comment]
> I found that explained on the runner-images repo [actions/runner-images#1918](https://github.com/actions/runner-images/issues/1918)

The description and comments make me feel like in that case this error was in actions not hitting timeout, but in my case the action seem to hit timeout (due to snapcraft hang)

![изображение](https://user-images.githubusercontent.com/17829319/223284665-b9fa3e35-4cfc-4cc3-a66f-9e39b3e63cae.png)

**Comment by nicolasbock** [comment]
> After successfully completing and snapping the action gets stuck (for a lack of a better word) and is **eventually timed out**. This looks like the snapcraft process is waiting on something it is not seeing.

@ilya-fedin no, the action times out in my case.

**Comment by ilya-fedin** [comment]
@nicolasbock I talk about the linked issue in the quote

**Comment by nicolasbock** [comment]
Ah, sorry, I misunderstood.

**Comment by ilya-fedin** [comment]
Downgrading action to Ubuntu 20.04 helps me
